client/client.go

```diff
@@ -20,9 +20,11 @@ type Context struct {
 }
 
 type Client interface {
-	TokenFromRequest(r http.Request) string
-
 	ActionAllowed(token string, action *Action) (*Context, error)
 
 	Authorized(token string, scopes ...string) (*Context, error)
```

Comment:
There will be no more middleware, just one client. This will clean up the code base a lot. This interface should basically be implemented by any client in any language. Maybe we can use something like Thrift to build a client library for various languages